### PR TITLE
Switch approach for staff dashboard access to org scoped URL

### PR DESCRIPTION
**Pull request by marcospri**
For:

- https://github.com/hypothesis/lms/issues/6509

- We have now duplicate entry points URL, the ones scoped to one org (for staff) and the rest (for the LMS users)

- The frontend fetches the revalvnt public_id from the path parameter instead of a js_config value

- API endoints still take a public_id as any other filtering option

### CODE REVIEW

File: lms/security.py [hunk 1]
```diff
@@ -99,11 +99,11 @@ def get_policy(request: Request):  # noqa: PLR0911
 
         path = request.path
 
-        if path.startswith("/admin") or path.startswith("/googleauth"):
+        if path.startswith(("/admin", "/googleauth", "/dashboard/org")):
```
Comment (acelaya) [inline lms/security.py:102]: I think it wouldn't make a difference, but perhaps this should check for `/dashboard/orgs`, with an `s` at the end?

```suggestion
        if path.startswith(("/admin", "/googleauth", "/dashboard/orgs")):
```

File: lms/static/scripts/frontend_apps/components/dashboard/CourseActivity.tsx [hunk 2]
```diff
@@ -32,6 +32,7 @@ type AssignmentsTableRow = {
  */
 export default function CourseActivity() {
   const { courseId } = useParams<{ courseId: string }>();
+  const { organizationPublicId } = useParams();
```
Comment (acelaya) [inline lms/static/scripts/frontend_apps/components/dashboard/CourseActivity.tsx:35]: You can merge these two calls to `useParams()`:

```suggestion
  const { courseId, organizationPublicId } = useParams<{ courseId: string; organizationPublicId?: string }>();
```

File: lms/static/scripts/frontend_apps/components/dashboard/AssignmentActivity.tsx [hunk 3]
```diff
@@ -27,6 +27,7 @@ type StudentsTableRow = {
 export default function AssignmentActivity() {
   const { dashboard } = useConfig(['dashboard']);
   const { routes } = dashboard;
+  const { organizationPublicId } = useParams();
   const { assignmentId } = useParams<{ assignmentId: string }>();
```
Comment (acelaya) [inline lms/static/scripts/frontend_apps/components/dashboard/AssignmentActivity.tsx:31]: You can merge these two calls to `useParams()`

```suggestion
  const { assignmentId, organizationPublicId } = useParams<{ assignmentId: string; organizationPublicId?: string }>();
```

**Review by acelaya — COMMENTED** [review]
Looks good in general. I can help with the FE failing test.

**Comment by acelaya** [comment]
I added https://github.com/hypothesis/lms/pull/6522, which fixes the failing FE test. Feel free to merge it.

**Review by acelaya — APPROVED** [review]
Looks good!

I just noticed a block of code that I think is not needed anymore and is probably being always skipped, but confirm this as I might be wrong.

File: lms/security.py [hunk 4]
```diff
@@ -99,11 +99,11 @@ def get_policy(request: Request):  # noqa: PLR0911
 
         path = request.path
 
-        if path.startswith("/admin") or path.startswith("/googleauth"):
+        if path.startswith(("/admin", "/googleauth", "/dashboard/orgs")):
             # Routes that require the Google auth policy
             return LMSGoogleSecurityPolicy()
 
-        if path.startswith(("/dashboard", "/api/dashboard")):
+        if path.startswith(("/api/dashboard")):
```
[the remark this answers is not in the document]
Comment (marcospri) [reply lms/security.py:106]: While the token is sent (and returned by the BE) that's not a valid token, well, it's empty, there's no LTI identity to pass around here.

We read the cookie instead. 

We could set the token to null and avoid having to send it in every request in the FE if you think that will make more clear what's going on.

[the remark this answers is not in the document]
Comment (marcospri) [reply lms/security.py:106]: See: https://github.com/hypothesis/lms/pull/6518/commits/08fb73dfe80874d38288191b8ae271749ab938df

**Pull request merged**